packages/react/src/components/select/custom/custom-select-portal.tsx

```diff
@@ -0,0 +1,116 @@
+import React, {
+  MutableRefObject,
+  ReactNode,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react';
+import { createPortal } from 'react-dom';
+
+export const CustomSelectPortal = ({
+  id,
+  children: { select, dropdown },
+}: CustomSelectPortalProps): JSX.Element => {
+  const target = usePortal(id);
+  const [ref, dimensions] = useDimensions();
+
+  target.style['position'] = 'fixed';
+
+  if (dimensions) {
+    target.style['top'] = `${dimensions.y + dimensions.height}px`;
+    target.style['left'] = `${dimensions.x}px`;
+    target.style['width'] = `${dimensions.width}px`;
+  }
+
+  return (
+    <>
+      {select(ref)}
+      {createPortal(dropdown, target)}
+    </>
+  );
+};
+
+export interface CustomSelectPortalProps {
+  children: {
+    dropdown: ReactNode;
+    select: (ref: MutableRefObject<HTMLElement | null>) => void;
+  };
+  id: string;
+}
+
+const usePortal = (id: string): HTMLDivElement => {
+  const ref = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const existingEl = document.querySelector(`#${id}`);
+    const element = existingEl || createElement(id);
+
+    if (!existingEl) addElement(element);
+    if (ref.current) element.appendChild(ref.current);
+
+    return () => {
+      ref.current?.remove();
+      if (!element.childElementCount) element.remove();
+    };
+  }, [id]);
+
+  // evaluate lazily, so that this runs only once and ref points to same element
+  // https://reactjs.org/docs/hooks-faq.html#how-to-create-expensive-objects-lazily
+  const getRootElem = () => {
+    if (!ref.current) ref.current = document.createElement('div');
+    return ref.current;
+  };
+
+  return getRootElem();
+};
+
+function createElement(id: string): HTMLDivElement {
+  const element = document.createElement('div');
+  element.setAttribute('id', id);
+  return element;
+}
+
+function addElement(el: Element): void {
+  document.body.appendChild(el);
+}
```

Comment:
I am not sure we should be creating the container outside of React's control, perhaps asking the user to "save a slot" for the Select dropdown container would be better?

I mean, the current way makes it much more convenient, I'm just worried we might run into other sorts of bugs, especially in MFE scenarios.

Reply:
This now adds a container for portal if not found, created with static ID "castor_custom_select_dropdown".

So, to avoid such, you'd only need to make sure a container with this ID is in the document.

We could treat this as an edge case, which would involve that extra step. Or even document as a solution.

Otherwise we can omit creating a container component, but still require such to be added manually when using a component.

We could also provide e.g. a CastorPortal component that is responsible to create all required containers, for Select and other components (like Tooltip if needed).

Reply:
Yea if that's the way we choose to go a component like `CastorPortal` would be very nice.

That's a breaking change, so might be good to choose now.

What do you think, should we require a Portal target like we do `Icons` for the sheet?

Reply:
Probably that would be best. Same pattern.

However, note that next release will already most likely be a 2.0 so we can do a breaking change.

Reply:
Right. Since we're not releasing "CustomSelect" until next release, we can work on that Portal component then.